### PR TITLE
enhancement(observability): Add internal events for `http` source

### LINKED ISSUE #3199: Define events for the `http` source
As part of finishing https://github.com/timberio/vector-product/issues/14, we need to define internal events for the `http` source.

## Defining events

* If you have not already, please read [RFC 2064](https://github.com/timberio/vector/blob/master/rfcs/2020-03-17-2064-event-driven-observability.md).
  * ***Be sure to read the [Guide-level Proposal section](https://github.com/timberio/vector/blob/master/rfcs/2020-03-17-2064-event-driven-observability.md#guide-level-proposal) and the implementation details that affect performance.***
* You can also look at the [currently defined events](https://github.com/timberio/vector/tree/master/src/internal_events) for precedence.
* Events should be named properly: `<ComponentName><Noun><Verb>`. For example, the `syslog` source defines the [`SyslogEventReceived` struct](https://github.com/timberio/vector/blob/fb71f5ad76266d9a44164e6804723376353e4ff0/src/internal_events/syslog.rs#L9).

## Requirements

- [ ] Add a `HTTPEventReceived` event.
  - [ ] Derive a `events_processed` metric via the `emit_metrics` method. ([example](https://github.com/timberio/vector/blob/master/src/internal_events/syslog.rs#L15))
  - [ ] If possible, derive a `bytes_processed` metric via the `emit_metrics` method. ([example](https://github.com/timberio/vector/blob/fb71f5ad76266d9a44164e6804723376353e4ff0/src/internal_events/syslog.rs#L19))
- [ ] Audit and add events for all possible errors. ([example](https://github.com/timberio/vector/blob/fb71f5ad76266d9a44164e6804723376353e4ff0/src/internal_events/syslog.rs#L31))
- [ ] Audit and add events for all component-specific behavior. ([example](https://github.com/timberio/vector/blob/fb71f5ad76266d9a44164e6804723376353e4ff0/src/internal_events/tcp.rs#L9)).
- [ ] Ensure that logs in the hot path are rate limited via the `rate_limit_secs` parameter. ([example](https://github.com/timberio/vector/blob/fb71f5ad76266d9a44164e6804723376353e4ff0/src/internal_events/tcp.rs#L83)).
- [ ] Ensure that all log messages start with a lowercase letter and end with a `.`. ([example](https://github.com/timberio/vector/blob/master/src/internal_events/tcp.rs#L83))

### PR DESCRIPTION
**Pull request by juchiast**
closes #3199 

Added `HTTPEventsReceived` and `HTTPBadRequest`.

### CODE REVIEW

File: src/sources/util/http.rs [hunk 1]
```diff
@@ -74,15 +76,20 @@ pub trait HttpSource: Clone + Send + Sync + 'static {
             .and(warp::path::end())
             .and(warp::header::headers_cloned())
             .and(warp::body::bytes())
-            .and_then(move |headers: HeaderMap, body| {
+            .and_then(move |headers: HeaderMap, body: Bytes| {
```
Comment (fanatid) [inline src/sources/util/http.rs:79]: Is this required change?

Comment (juchiast) [reply src/sources/util/http.rs:79]: yes, rustc couldn't infer the type when I add `let body_size = body.len();`.

File: src/internal_events/http.rs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+#[derive(Debug)]
+pub struct HTTPEventsReceived {
+    pub events_count: usize,
+    pub byte_size: usize,
+}
+
+impl InternalEvent for HTTPEventsReceived {
+    fn emit_logs(&self) {
+        debug!(
```
Comment (fanatid) [inline src/internal_events/http.rs:12]: I'm not completely sure, but I think this should be `trace!`. #3285

Comment (binarylogic) [reply src/internal_events/http.rs:12]: Yeah, this one is a little different given that it's processing a batch of events. @juchiast let's move it to `trace!` for now.

Comment (juchiast) [reply src/internal_events/http.rs]: Changed to trace!

**Pull request merged**